src/core/imap/MCIMAPSession.cpp

```diff
@@ -380,6 +381,14 @@ IMAPSession::~IMAPSession()
 void IMAPSession::setHostname(String * hostname)
 {
     MC_SAFE_REPLACE_COPY(String, mHostname, hostname);
+    
+    if(!mNetEaseWorkaroundEnabled && hostname 
```

Comment:
are you matching against the email address or hostname?
(Below, you're matching with the @ sign).